sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_converters/_ai_services.py

```diff
@@ -0,0 +1,468 @@
+
+        :param thread_id: The ID of the thread.
+        :type thread_id: str
+        :return: A list of messages in chronological order.
+        :rtype: List[Message]
+        """
+        messages = self.project_client.agents.list_messages(
+            thread_id=thread_id, limit=100, order=ListSortOrder.ASCENDING
+        )
+        return messages.data
+
+    def __list_tool_calls_chronological(self, thread_id: str, run_id: str) -> List[ToolCall]:
+        """
+        Lists tool calls in chronological order for a given thread and run.
+
+        :param thread_id: The ID of the thread.
+        :type thread_id: str
+        :param run_id: The ID of the run.
+        :type run_id: str
+        :return: A list of tool calls in chronological order.
+        :rtype: List[ToolCall]
+        """
+        # This is the other API request that we need to make to AI service, such that we can get the details about
+        # the tool calls and results. Since the list is given in reverse chronological order, we need to reverse it.
+        run_steps_chronological: List[RunStep] = self.project_client.agents.list_run_steps(
+            thread_id=thread_id, run_id=run_id, limit=100, order=ListSortOrder.ASCENDING
+        ).data
+
+        # Let's accumulate the function calls in chronological order. Function calls
+        tool_calls_chronological: List[ToolCall] = []
+        for run_step_chronological in run_steps_chronological:
+            if run_step_chronological.type != _TOOL_CALLS:
+                continue
+            step_details: RunStepToolCallDetails = run_step_chronological.step_details
+            if step_details.type != _TOOL_CALLS:
+                continue
+            if len(step_details.tool_calls) < 1:
+                continue
+            for tool_call in step_details.tool_calls:
+                # We need to add the tool call and the result as two separate messages.
+                tool_calls_chronological.append(
+                    ToolCall(
+                        created=run_step_chronological.created_at,
+                        completed=run_step_chronological.completed_at,
+                        details=tool_call,
+                    )
+                )
+
+        return tool_calls_chronological
+
+    def __list_run_ids_chronological(self, thread_id: str) -> List[str]:
+        """
+        Lists run IDs in chronological order for a given thread.
+
+        :param thread_id: The ID of the thread.
+        :type thread_id: str
+        :return: A list of run IDs in chronological order.
+        :rtype: List[str]
+        """
+        runs = self.project_client.agents.list_runs(thread_id=thread_id, order=ListSortOrder.ASCENDING)
+        run_ids = [run["id"] for run in runs["data"]]
+        return run_ids
+
+    @staticmethod
+    def __extract_tool_definitions(thread_run: ThreadRun) -> List[ToolDefinition]:
+        """
+        Extracts tool definitions from a thread run.
```

Comment:
Shall we update it to say it gives only Function Tool Defintition ?

Reply:
Good pick, I've added an explanation with a TODO outlining the current limitation in AI Services API. Renamed it to `__extract_function_tool_definitions`.